core/block_svg.js

```diff
@@ -1061,75 +1053,35 @@ Blockly.BlockSvg.prototype.dispose = function(healStack, animate) {
   this.svgGroup_ = null;
   this.svgPath_ = null;
   this.svgPathLight_ = null;
-  this.svgPathDark_ = null;
   Blockly.utils.dom.stopTextWidthCache();
 };
 
 /**
  * Change the colour of a block.
+ * @package
  */
-Blockly.BlockSvg.prototype.updateColour = function() {
+Blockly.BlockSvg.prototype.applyColour = function() {
   if (!this.isEnabled()) {
     // Disabled blocks don't have colour.
     return;
   }
 
-  if (this.isShadow()) {
-    this.setShadowColour_();
-  } else {
-    this.setBorderColour_();
-    this.svgPath_.setAttribute('fill', this.getColour());
-  }
+  this.pathObject.applyColour(this.isShadow());
 
   var icons = this.getIcons();
   for (var i = 0; i < icons.length; i++) {
-    icons[i].updateColour();
+    // TODO: Decide whether to make icon and field applyColour take in a
```

Comment:
I'm open to suggestions here.  I'll make an actual issue if it's a change that we want.

Also, note that I changed a name: `field.updateColour` -> `field.applyColour`. 

This was to make names match, and to indicate that the colour model isn't changing, it's that the change is being applied to the rendered block.  I could be talked into changing it back if we want to leave the name alone for field.